strapi/strapi
# ML: Apply fallback mime-type if none is set

### What does it do?

While debugging https://github.com/strapi/strapi/issues/12761 we discovered that koa sometimes fails to set a mime-type for uploaded files through the API. Instead, it assumes `null` and is written to the database, which causes the admin panel to crash, because we rely on the presence of a mime-type in several cases (e.g. to different components for different assets). A working python reproduction can be found here: https://github.com/gu-stav/strapi-upload-rest-graphql

This PR fixes the root cause, however, we might need more work to fully fix the issue, since users might have a corrupted database state. We could therefore create a migration. If we decide on migration the frontend can stay as it is. If not, we need to handle the case, that an asset doesn't have a mime-type and display the default component for it.

This PR also fixes the issue, that files uploaded through `/graphql` are not cleaned up from the temporary folder.

What do you think?

### Why is it needed?

To set the mime-type of a file more reliable.

### How to test it?

Once we agree this is a valid fix for the issue we should add e2d tests. This might be hard because we couldn't find a way to send a file without a mime-type in JavaScript which might be happening, because we are using `FormData`.

### Related issue(s)/PR(s)

- Fixes: https://github.com/strapi/strapi/issues/12761
- Fixes: https://github.com/strapi/strapi/issues/13174
- Fixes: https://github.com/strapi/strapi/issues/12897
